### PR TITLE
[WIP][5.2] Add UploadedFile interface

### PR DESCRIPTION
**Pull request by hasyimibhar**
Refer to issue #12432. Please advise if this is suitable for 5.3 instead.

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
This makes no sense because the interface is never used in the code.

**Comment by GrahamCampbell** [comment]
I just saw your issue where you talk about mocking. You really do need to update all instanceof checks and the like inside laravel or this won't work at all.

**Comment by hasyimibhar** [comment]
I added `size()`, `mimeType()`, and `isValid()` methods to the interface so that `Illuminate\Validation\Validator` can depend on the interface.

**Comment by GrahamCampbell** [comment]
Modifying the existing interfaces can only be done on 5.3 for BC reasons.

File: src/Illuminate/Validation/Validator.php [hunk 1]
```diff
@@ -15,9 +15,8 @@
 use Illuminate\Support\Fluent;
 use Illuminate\Support\MessageBag;
 use Illuminate\Contracts\Container\Container;
-use Symfony\Component\HttpFoundation\File\File;
 use Symfony\Component\Translation\TranslatorInterface;
-use Symfony\Component\HttpFoundation\File\UploadedFile;
+use Illuminate\Contracts\Http\UploadedFile;
```
Comment (GrahamCampbell) [inline src/Illuminate/Validation/Validator.php]: Please sort imports by length.

**Comment by hasyimibhar** [comment]
> Modifying the existing interfaces can only be done on 5.3 for BC reasons.

Okay, moving this to 5.3. Refer to PR #12463.

**Pull request closed without merging**